cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -40,11 +40,40 @@ type MasterConfiguration struct {
 	KubernetesVersion string
 	CloudProvider     string
 	AuthorizationMode string
+
+	Phases Phases
+
+	// The directory where certificates are stored
+	CertificatesDir string
+}
+
+type PhaseMetadata struct {
+	Annotations map[string]string
+}
+
+type Phases struct {
+	Certificates CertificatesPhase
+}
+
+type CertificatesPhase struct {
+	Metadata PhaseMetadata
+
+	// In the future, we may provide more options for generating certs
+	// For instance, we may want to provide integrations with something like Vault in the future for storing the certs
+	SelfSign *SelfSignCertificates
+}
+
+type SelfSignCertificates struct {
```

Comment:
`ca.{crt,key}`, `apiserver.{crt,key}`, `apiserver-kubelet-client.{crt,key}` and soon `sa.{crt,key}`
If some of the files already exist, they won't be overridden, just validated.